src/platform/Ameba/AmebaOTAImageProcessor.cpp

```diff
@@ -327,26 +333,63 @@ void AmebaOTAImageProcessor::HandleProcessBlock(intptr_t context)
         else
         {
             ChipLogError(SoftwareUpdate, "Invalid size");
+            imageProcessor->mDownloader->EndDownload(CHIP_ERROR_WRITE_FAILED);
             return;
         }
     }
 #endif
-    imageProcessor->mParams.downloadedBytes += imageProcessor->mBlock.size();
+    imageProcessor->mParams.downloadedBytes += block.size();
     imageProcessor->mDownloader->FetchNextData();
 }
 
 void AmebaOTAImageProcessor::HandleApply(intptr_t context)
 {
     auto * imageProcessor = reinterpret_cast<AmebaOTAImageProcessor *>(context);
+    if (imageProcessor == nullptr)
+    {
+        return;
+    }
+
+    OTARequestorInterface * requestor = chip::GetRequestorInstance();
+    if (requestor != nullptr)
+    {
+        // TODO: Use software version from Configuration Manager
+        requestor->NotifyUpdateApplied(imageProcessor->mSoftwareVersion);
+    }
 
     // Reboot
     ota_platform_reset();
 }
 
+CHIP_ERROR AmebaOTAImageProcessor::ProcessHeader(ByteSpan & block)
+{
+    ChipLogProgress(SoftwareUpdate, "ProcessHeader");
+    if (mHeaderParser.IsInitialized())
+    {
+        OTAImageHeader header;
+        CHIP_ERROR error = mHeaderParser.AccumulateAndDecode(block, header);
+
+        // Needs more data to decode the header
+        ReturnErrorCodeIf(error == CHIP_ERROR_BUFFER_TOO_SMALL, CHIP_NO_ERROR);
+        ReturnErrorOnFailure(error);
+
+        // We save the software version to be used in the next NotifyUpdateApplied, but it's a non-standard
```

Comment:
Are you sure you wanted copy that functionality?

Reply:
This has been removed.